Modify Steam sign-in button to comply with Valves new OpenID Requests

JasonRivers opened the pull request:
Possibly due to the old button not saying "not associated with Valve"
New icons with this text have been created (probably to stop more law suits).

This replaces the sign in through STEAM button with the new one.

JasonRivers commented:
This pr is ready to go, it is only an update to the steam login image button to conform to valves TOS. The change came about because of the gambling sites, the new buttons say that the site is not affiliated with valve.

meklu commented:
This doesn't seem to fill the whole vertical height of the nav bar, and it breaks some other parts of the looks as well. This would either need some CSS tweakage or an adaptation of the new image to e.g. the dimensions of the old one.

JasonRivers commented:
Just updated the PR with new CSS, the burger menu is invoked slightly earlier now due to the size of the steam logo, I also noticed there was some padding on the steam login li when the burger menu is active (on the live site) I have fixed this as part of this work.

meklu commented on css/bootstrap.steamlug.css:
```diff
@@ -1026,7 +1026,7 @@ p a.group-join{font-size:100%}
 	from{width:0}
 }
-.navbar-nav>li.steamLogin{right:0;padding-right:11px}
+.navbar-nav>li.steamLogin{right:0;padding-right:11px;margin-top:4px}
```
Instead of adding a margin here, you should probably adjust the padding on the next line.

    padding:<top> <right> <bottom> <left>;

The bar is 50px high apparently judging by the rest of our CSS, so the rule on line 1030 would become this when taking your 4px adjustment into account:

    padding:7px 5px 8px 3px;

JasonRivers commented:
@meklu done, I also adjusted the padding so the left & right are even as this shows in black on a hover, it looks better when you hover over it if it's even.

her001 commented:
If no other corrections are needed, I think you should squash the last two commits into one before a merge, to keep the CSS tweaks together.

meklu commented:
Looks good to me!

meklu commented:
If you take a look at https://staging.steamlug.org/, the horizontal padding needs a little trimming (a ≤ 7px total seems to work here) or the width needs to be adjusted.

meklu commented:
I might just push meklu/steamlug.org@1b6341efe20d47f35ef838178ef32f433500b5c9.

johndrinkwater commented:
```
-@media(min-width:768px){
+@media(min-width:835px){
```
It would be best to avoid changing site-wide minimum (and maximum) width for the whole page, just to be able to update this image. They are set sizes picked for good reasons for a mixture of devices.

Maybe a better option (for looks and layout) would be to copy what dotabuff has done and just use the Steam logo next to ‘Sign in’ text.

![screenshot 2017-04-24 150118](https://cloud.githubusercontent.com/assets/7708/25340964/da41ae3c-28f6-11e7-97c0-74963fdbe92d.png)

That would involve adding the Steam icon to our custom font, and poking some HTML.

her001 commented:
Just testing https://staging.steamlug.org, it seems to display fine on an Android device with a 720px wide screen and in Firefox's responsive design mode set to 280px wide. Chromium's device mode seems entirely broken, but a Chromium window seems to render fine until it gets to below about 350px wide.

The reason this was chosen is because Valve requests that one of two images they provide are used.

From their [dev page](https://steamcommunity.com/dev):

> If you are using OpenID on your site, we request that you use one of the following buttons as your link to the Steam sign in page.

johndrinkwater commented:
Yes, I know. I did not imagine that this ugly button was made for this PR.

It is their request, not a demand or a requirement. We don’t even need the original sits image! I request that we don’t have an uglier button than the one we have now.

Corben78 commented:
Imho there is no need to have the text "not associated with valve" in the button.
There is a big orange warning on steam's auth page, which should be enough hint for everybody.
I would prefer the former button, it was already changed from green background to black background, which fits pretty well to the current design.
Or, where john already hinted at, use a self made button, maybe with the actual steam logo instead.

JasonRivers commented:
@johndrinkwater @Corben78 While we might not feel that it needs the text under the button, Valve have changed the buttons for good reason. Conforming to their request is safer than not IMHO. Even before Valve changed the button, the steam log in page still had the orange banner, but they still decided they needed to change the button too.

johndrinkwater commented:
Valve did this to further push liability onto their users. Not only does it change nothing legally for most Europeans and other international users, it is just fecking ugly.

I would be less bothered if it were pretty or the same size as before, but instead https://www.reddit.com/r/CrappyDesign/comments/4mx4g2/valves_new_official_sign_in_through_steam_buttons/

@Corben78 ’s previous change imo was perfect, making the sits button better fit the theme https://github.com/SteamLUG/steamlug.org/commit/97d9d3b089b4b52b7ab4e602a452d774911aa988

her001 commented:
While I agree that the new one is not as pretty, I didn't think it was that big of a deal.

This PR has sat here for over nine months without anyone publicly disputing the use of this image, then merged before it was brought up. Maybe the best place for this discussion would be in a new PR containing a revert commit for 8ce0123, 4ba5cbf, and 1b6341e.

JasonRivers commented:
@johndrinkwater I think you misunderstand. I don't disagree that the buttons are fugly, They are. I'm simply stating that Valve intentionally added the text and removed the steam logo on those buttons for good reason. I'm sure we could come up with our own buttons, but I currently do not understand the legal consequences of doing so, and there-for, having the button that Valve requests people use is the safest option.
If you do know the legal consequences of either using our own, or modifying theirs, please let me know (I am actually interested in this).

johndrinkwater commented:
> This PR has sat here for over nine months

Understand that not merging was an active action, pull **requests** can sit 😜

> I don't disagree that the buttons are fugly, They are

Then why did you make a commit for a button that looks more unofficial than the one that already is? Few sites have moved to this new form _because_ imo it looks shadier.

http://steamcommunity.com/dev/apiterms is our agreement & terms to use the API, which doesn’t strictly cover OpenID logins except for this `The Steam Web API may require a licensed Steam end user to authenticate a Steam account. You will abide by the Steam Web API documentation and, in particular, you will not intercept or store the end user's Steam password on log in.` which we do not.
Since we already abide by `You may not present the Steam Data (or permit the Steam Data to be presented) so that it appears (a) that your Application is endorsed or affiliated with Valve or Steam, or (b) to be available from a third party. `, this button update seems redundant.

I don’t want to be passionate about a bloody button,  just want to know that people are passionate about this change being accepted onto master. If you’re not, I would like it reverted.

her001 commented:
> Understand that not merging was an active action, pull requests can sit 😜

After many calls for review on irc and lots of time passing, nobody objected to this. Nobody left a comment at all for most of the time, in fact. Eventually, @meklu provided some constructive feedback, which was quickly acted upon, then the PR was merged.

Besides, letting a PR sit without feedback is a great way to discourage potential contributors from working with us.

JasonRivers commented:
@johndrinkwater you seem to be passionate about this, I apologise for any controversy I have caused, However, I do have to agree with @her001 that a comment on the PR wouldn't have gone a miss months ago.

<s> Unfortunately, I'm not very good at the art of telepathy. </s>

Corben78 commented:
As already stated, I would prefer the old button back. Especially as there doesn't seem to be any legal issues, where valves forces to use their proposed button.

sorry @JasonRivers ;)

johndrinkwater commented:
Our pull requests are open for anyone to submit anything they think would benefit us, there should never be an expectation that we accept everything. I admit that I should have closed this earlier, or said I think it is an ungood idea. _Or anybody else could have done._

her001 commented:
If anyone said that they didn't like it, closed it, or even put a thumbs down on it, I would be a lot less annoyed at the sudden movement to revert this. As I have mentioned, this has been open for a long time with no comment, even after many requests for review in IRC.

> Or anybody else could have done.

It seems that only two people dislike this change. Everyone else was okay with it, so why would they say they didn't like it or close the issue?
